app/controllers/VatDetailsController.scala

```diff
@@ -59,16 +60,24 @@ class VatDetailsController @Inject()(val messagesApi: MessagesApi,
         nextReturn <- returnObligationsCall
         nextPayment <- if (retrieveHybridStatus(customerInfo)) Future.successful(Right(None)) else paymentObligationsCall
         mandationStatus <- if(appConfig.features.submitReturnFeatures()) {
-          mandationStatusService.getMandationStatus(user.vrn) } else { Future.successful(Right(MandationStatus("Disabled"))) }
+          retrieveMandationStatus(user.vrn)
+        } else { Future.successful(Right(MandationStatus("Disabled"))) }
       } yield {
         val migratedDate = (request.session.get(SessionKeys.migrationToETMP), customerInfo) match {
           case (Some(date), _) => date
           case (None, Right(details)) => details.customerMigratedToETMPDate.getOrElse("")
           case (None, Left(_)) => ""
         }
         auditEvents(user, nextReturn, nextPayment)
+
+        val newSessionVariables: Seq[(String, String)] = Seq(SessionKeys.migrationToETMP -> migratedDate) ++ (mandationStatus match {
+          case Right(status) => Seq(SessionKeys.mandationStatus -> Json.stringify(Json.toJson(status)))
```

Comment:
Please could you store mandationStatus in the session as a String instead of JSON? Makes it simpler and follows same pattern as the other two